src/starkware_utils/commitment_tree/binary_fact_tree_node.rs

```diff
@@ -0,0 +1,322 @@
+use std::collections::HashMap;
+use std::marker::PhantomData;
+
+use futures::future::FutureExt;
+use num_bigint::BigUint;
+
+use crate::starkware_utils::commitment_tree::base_types::{Height, TreeIndex};
+use crate::starkware_utils::commitment_tree::binary_fact_tree::BinaryFactDict;
+use crate::starkware_utils::commitment_tree::errors::TreeError;
+use crate::starkware_utils::commitment_tree::inner_node_fact::InnerNodeFact;
+use crate::starkware_utils::commitment_tree::leaf_fact::LeafFact;
+use crate::starkware_utils::commitment_tree::merkle_tree::traverse_tree::{traverse_tree, TreeTraverser};
+use crate::storage::storage::{FactFetchingContext, HashFunctionType, Storage, StorageError};
+
```

Comment:
Some comments or pointers in general in this file would help, or perhaps keeping things `pub(crate)` if they aren't meant to be used externally...